bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -341,12 +346,43 @@ private int readFromLogChannel(long entryLogId, BufferedReadChannel channel, Byt
         }
     };
 
+    private final  CacheLoader<Long, FileChannel> loader = new CacheLoader<Long, FileChannel> () {
```

Comment:
I would suggest renaming this to a more meaningful name, like `readonlyFileChannelLoader`.